mantidproject/mantid
# Save user file name in SaveCanSas

*#28513 Must be merged first*

**Description of work.**

Adds steps to save out the Mask File in SaveCanSAS and adds a warning if there is none associated with the workspace

**Report to:** ISIS / Steve K

**To test:**
- Download the ISIS Training Data again (even if you have it) as it's been updated
- Open the SANS GUI
- Open the TOML file with `MaskFile.toml`
- Set the batch file to batch.csv
- At the bottom ensure either `File` or `Both` is checked, and make sure `SaveCanSas` is also ticked
- Process any workspace
- Open the resulting `.xml` file in your save directory, and check the `user_file` tag is correct

Fixes #28846 and #28847 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
